src/scheduler.py

```diff
@@ -133,14 +148,46 @@ def log_route(
         Writes a log to the standard logger and to the log file of a route.
         """
 
-        route.logs.append(f"{log_level.upper()} {fmt_string % tuple(args)}")
+        def asset_balance_prefix(leg: Leg, asset: str) -> Optional[str]:
+            balance_resp_asset = try_multiple_clients(
+                self.clients[leg.backend.chain_id],
+                lambda client: client.query_bank_balance(
+                    Address(
+                        self.wallet.public_key(),
+                        prefix=leg.backend.chain_prefix,
+                    ),
+                    asset,
+                ),
+            )
+
+            if not balance_resp_asset:
+                return None
+
+            return f"balance[{leg.backend.chain_id}]({asset[:DENOM_BALANCE_PREFIX_MAX_DENOM_LEN]}): {balance_resp_asset}"
+
+        def leg_balance_prefixes(leg: Leg) -> list[str]:
+            assets = [leg.in_asset(), leg.out_asset()]
+
+            return [
+                x for x in (asset_balance_prefix(leg, asset) for asset in assets) if x
```

Comment:
would be nice to get some comments around oneliners like this

Reply:
Done!